src/propertySelector/propertySelector.js

```diff
@@ -0,0 +1,150 @@
+     * @param {*} property
+     * @returns JQuery element containing property description
+     */
+    function createPropertyOption(property) {
+        const descriptionData = Object.assign({}, property);
+        if (search !== '') {
+            descriptionData.label = highlightCharacter(descriptionData.label, search, search);
+        }
+        const $propertyDescription = $(propertyDescriptionTpl({ property: descriptionData }));
+        const $checkboxContainer = $('.checkbox-container', $propertyDescription);
+        $checkboxContainer.append(checkBoxTpl({ id: descriptionData.id, checked: descriptionData.selected }));
+        const $checkbox = $('input', $checkboxContainer);
+        $checkbox.on('change', function () {
+            if (this.checked) {
+                selectedProperties.add(property.id);
+            } else {
+                selectedProperties.delete(property.id);
+            }
+        });
+        return $propertyDescription;
+    }
+
+    const instance = component({
+        positionContainer: function positionContainer() {
+            let { top, left, right, bottom } = this.config.data.position;
+            let maxHeight;
+            if (typeof bottom === 'undefined') {
+                maxHeight = $container.parent().height() - top - parentGap;
+            }
+            if (typeof top === 'undefined') {
+                maxHeight = $container.parent().height() - bottom - parentGap;
+            }
+            $container.css({ top, left, right, bottom, maxHeight });
+        },
+
+        /**
+         * Updates the list
+         */
+        redrawList: function redrawList() {
+            $propertyListContaner.empty();
+            availableProperties.forEach(property => {
+                property.selected = selectedProperties.has(property.id);
+                if (search === '' || property.label.toLowerCase().includes(search.toLowerCase())) {
+                    $propertyListContaner.append(createPropertyOption(property));
+                }
+            });
+        },
+        /**
+         * Adds and setups buttons to button container
+         */
+        addButtons: function addButtons() {
+            const cancelButton = buttonFactory({
+                id: 'cancel',
+                label: 'Cancel',
+                type: 'info',
+                cls: 'btn-secondary'
+            }).on('click', () => {
+                this.trigger('cancel');
+            });
+
+            const saveButton = buttonFactory({
+                id: 'save',
+                label: 'Save',
+                type: 'info'
+            }).on('click', () => {
+                this.trigger('select', [...selectedProperties]);
+            });
+
+            cancelButton.render($buttonsContainer);
+            saveButton.render($buttonsContainer);
+        },
+        /**
+         * Setups search input event listners
+         */
+        setupSearch: function setupSearch() {
```

Comment:
**suggestion:** 
```suggestion
        setupSearch() {
```


**thought:** Same consideration here. This is a setup function, it should not be an exposed API. It will be called only once from the render step, which is unique in the component life-cycle. There is no benefit for the consumer to access it. Moreover, it may be harmful to call it arbitrarily from the outside.